batch2/batch/front_end/front_end.py

```diff
@@ -345,9 +346,13 @@ def create_job(app, jobs_builder, batch_id, userdata, parameters):  # pylint: di
     user = userdata['username']
     batch = await Batch.from_db(request.app, batch_id, user)
     if not batch:
-        abort(404)
-    asyncio.ensure_future(batch.mark_deleted())
-    return jsonify({})
+        raise web.HTTPNotFound()
+    async with aiohttp.ClientSession(
+            raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
+        async with session.delete(
```

Comment:
Does this need a retry of some sort? What happens if it fails? Same question applies for cancel and close. Also authentication of request? I'm fine if these are to-do items.

Reply:
Yes, I'm very concerned about failures/retries.  I think there are issues related to this throughout our code (the recent auth change was only one step) and I'm formulating a discipline for errors and retries that I will apply throughout including in this code.  Let's call it a to-do item.

Yes, authentication was noted as a to-do item in your original PR, which we're still using to track work as per @danking's suggestion.